docs/src/main/tut/user-guide.md

```diff
@@ -803,6 +803,115 @@ implicit val encodeExceptionCirce: Encoder[Exception] = Encoder.instance(e =>
 If no other `Encode[Exception]` is available, Finch provides a fallthrough of `Encode.Aux[Exception, ?]`
 that will return an empty content body.
 
+### Streaming
+
+`finch-iteratee` module enables high-level support of chunked request and response streaming using
+[iteratee.io](https://github.com/travisbrown/iteratee) and `circe-streaming` libraries. It allows to encode and decode
+newline delimeted JSON streams, but also supports binary and text streaming.  
+Concept of [iteratee](https://en.wikipedia.org/wiki/Iteratee) could be complicated for understanding, but
+proves itself very powerful and useful abstraction over processing sequential data. API of `iteratee.io` library is transparent and covers most of 
+the potential use cases.
+
+Main points:
+
+- `Enumerator` is a "lazy storage" with all the chunks that already were received or about to be received in a future
+- `Enumeratee` is a `map` part and could be used to transform chunks in `Enumerator`
+- `Iteratee` is a `reduce` part that runs computation and processes over chunks in `Enumerator`
+
+**Decoding**
+
+Currently JSON decoding is supported only in `finch-circe` module and implemented
+using `circe-streaming` library.
+
+To make it happen finch out of the box provides few instances for `io.finch.iteratee.Enumerate` abstraction:
```

Comment:
How about:

Finch plugs in iteratee-powered decoding support via the `io.finch.iteratee.Enumerate` type-class.